python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -331,22 +309,25 @@ def __getitem__(self, this_id):
             this_id (int): This is the key of an object in the object dictionary
         """
         logger.debug("__getitem__")
+        # Is this an Incomplete Object?
+        if this_id in self._incomplete_objects:
+            return IncompleteObject(self, this_id)
+
+        # Nope, so try to find it and raise an exception if not
         try:
             return self._objects[this_id]
         except KeyError as err:
             logger.debug("Repo KeyError: %s" % err)
             logger.debug("Keys: %s id: %s" % (self._objects.keys(), this_id))
-            if this_id in self._incomplete_objects:
```

Comment:
I'm not sure I understand - If a load fails then the object is added to `_incomplete_objects`. Then, the next time it's referenced it will be picked up in the first part of the code and an `IncompleteObject` will be returned. The only reason this code will return a `KeyError` is if the object isn't in **either** `_incomplete_objects` or `_objects` in which case, that would be expected wouldn't it?